### PR TITLE
Digest algorithms

### LINKED ISSUE #364: blake2b algorithm identifier
`blake2b` is listed in the digest algorithm table, but the output size is not specified. The notes say "full-length form only," which I assume means `blake2b-512`.  For the sake of clarity and interpretability, I think it would be useful to use the complete identifier.

### LINKED ISSUE #371: Make it clear that allowed digest algorithms are a controlled vocab
In the "Digests" section at: https://ocfl.io/0.3/spec/#digests

It identifies two uses for digests, "content-addressing" and "fixity-values". 

It says you "SHOULD use sha512" for content-addressing.

It says you "MAY choose from the following list of digest algorithms" for fixity-values. 

1) I believe the intent is that for both content-addressing and fixity-values you MUST NOT use anything _but_ what is in this list. But that's not clear from those two statements, I think it needs to be clearer. 
    * For content-addressing, it just says you SHOULD use sha512 for content-addressing, but doesn't explicitly say that if use something other than sha512 it MUST be from that list. But I think that is intended. 
    * For fixity, It could be interpreted as you "MAY" use something from the list but could use anything else as well. I think it means you "MAY" have additional fixity values _or not_, but if you do it MUST be a digest from the list. 

2) I believe the intent is that the left column in the table of digest algorithms is a controlled vocabulary of values for you to use in the `digestAlgorithm` field for content-addresses, or as the keys in the hash structures for `fixity` section. But I don't think any part of the spec actually clearly says this, and it should. 
    * This is also what confirms to me 1 above, that you MUST use only digest algorithsm from this table -- becuase if you choose some other arbitrary digest algorithm, what would you use as the value of `digestAlgorithm` and/or the keys for `fixity` section hashes, and how would we ensure that people were using the same strings for the same algorithms, and that it would be understood by a client?

### PR DESCRIPTION
**Pull request by ahankinson**
This PR fixes some language about the intended use of the table of digest algorithms as a controlled vocabulary. 

Also clarifies the length of `blake2b` to be 512. 

Fixed #371 
Fixed #364 
Refs #292

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -580,8 +581,7 @@ <h2>Basic Structure</h2>
                 <dt><code>digestAlgorithm</code></dt>
                 <dd>
                     The digest algorithm used for calculating digests within the OCFL Object. This SHOULD be
-                    <code>sha512</code>, however other values are permitted. See the <a href="#digests">section on
-                    Digests</a> for more information.
+                    <code>sha512</code>, however <code>sha256</code> MAY also be used.
```
Comment (awoods) [inline draft/spec/index.html]: Keeping:
```
See the <a href="#digests">section on Digests</a> for more information.
```
...still seems helpful.

File: draft/spec/index.html [hunk 2]
```diff
@@ -734,10 +734,11 @@ <h2>Fixity</h2>
                 key of <code>fixity</code> within the inventory.
             </p>
             <p>
-                The <code>fixity</code> block MUST contain keys corresponding to <a href="#digest-algorithms">digest
-                algorithms listed in the Digests section</a>. The value of each key MUST follow the structure of the
-                <a href="#manifest"><code>manifest</code></a> block; that is, a key corresponding to the digest value,
-                and an array of <a>content path</a>s that match that digest.
+                The <code>fixity</code> block SHOULD contain keys corresponding to the controlled vocabulary given in
+                the <a href="#digest-algorithms">digest algorithms</a> listed in the Digests section. Additional keys
+                MAY be used if they are supplied in an extension. The value of each key MUST follow the structure of
```
Comment (awoods) [inline draft/spec/index.html]: Include reference to external `extension`.

File: draft/spec/index.html [hunk 3]
```diff
@@ -474,7 +474,8 @@ <h2>Digests</h2>
         </p>
         <p>
             For storage of additional fixity values, and perhaps to support legacy content migration, implementers
-            MAY choose from the following list of digest algorithms:
+            SHOULD choose from the following controlled vocabulary of digest algorithms. Additional algorithms MAY be
+            supplied using an extension mechanism, with no expectation that all clients will support them.
```
Comment (awoods) [inline draft/spec/index.html]: Include reference to external `extension`.

File: draft/spec/index.html [hunk 4]
```diff
@@ -734,10 +744,11 @@ <h2>Fixity</h2>
                 key of <code>fixity</code> within the inventory.
             </p>
             <p>
-                The <code>fixity</code> block MUST contain keys corresponding to <a href="#digest-algorithms">digest
-                algorithms listed in the Digests section</a>. The value of each key MUST follow the structure of the
-                <a href="#manifest"><code>manifest</code></a> block; that is, a key corresponding to the digest value,
-                and an array of <a>content path</a>s that match that digest.
+                The <code>fixity</code> block SHOULD contain keys corresponding to the controlled vocabulary given in
+                the <a href="#digest-algorithms">digest algorithms</a> listed in the Digests section. Additional keys
+                MAY be used if they are supplied in an <a href="">extension</a>. The value of each key MUST follow the
```
Comment (rosy1280) [inline draft/spec/index.html]: this href may not work.

**Pull request merged**